spring-web/src/main/java/org/springframework/http/client/reactive/ApacheClientHttpConnector.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.http.client.reactive;
+
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+import org.apache.hc.client5.http.cookie.BasicCookieStore;
+import org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient;
+import org.apache.hc.client5.http.impl.async.HttpAsyncClients;
+import org.apache.hc.client5.http.protocol.HttpClientContext;
+import org.apache.hc.core5.http.HttpResponse;
+import org.apache.hc.core5.http.Message;
+import org.apache.hc.core5.http.nio.support.BasicRequestProducer;
+import org.apache.hc.core5.reactive.ReactiveEntityProducer;
+import org.apache.hc.core5.reactive.ReactiveResponseConsumer;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.core.io.buffer.DataBufferFactory;
+import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.http.HttpMethod;
+import org.springframework.lang.Nullable;
+
+/**
+ * {@link ClientHttpConnector} implementation for the Apache HttpComponents HttpClient 5.x.
+ *
+ * @author Martin Tarjányi
+ * @since 5.3
+ * @see <a href="https://hc.apache.org/index.html">Apache HttpComponents</a>
+ */
+public class ApacheClientHttpConnector implements ClientHttpConnector {
```

Comment:
I'd argue that `Apache` HTTP Client is more recognizable for users than `HttpComponents` HTTP client. Consistency is a strong point, though.

Reply:
It's mostly for consistency, but I also think that Apache is too broad a term, covering everything from the original httpd to the Java code that used to be known as Jakarta (which includes the HTTP client). By prefixing it with Apache, we would rule out any other possible connectors coming from the Apache project.

Reply:
@poutsma @martin-tarjanyi Is there any chance you could call those classes `ApacheHC*` instead of `HttpComponents*`. Or `ApacheHttpComponents*` worst case?

Reply:
We have decided to stick with the `HttpComponents` prefix for the class names, mainly for consistency with existing code (i.e. `HttpComponentsClientHttpRequestFactory`), but also because we do not use the `Apache` prefix for other classes that offer support for Apache projects (i.e. `TomcatHeadersAdapter`, `CommonsLogWriter`, `GroovyMarkupView`, etc.)